handler/istorehandler.go

```diff
@@ -0,0 +1,29 @@
+// Copyright 2024 Contributors to the Veraison project.
+// SPDX-License-Identifier: Apache-2.0
+package handler
+
+import (
+	"github.com/veraison/services/plugin"
+	"github.com/veraison/services/proto"
+)
+
+// IStoreHandler defines the interface to functionality for working with
+// attestation scheme specific store interfaces. This includes extracting
+// Trust Anchor IDs from attestation token, and synthesizing,
+// Reference Value and TrustAnchor Keys from supplied endorsements
+type IStoreHandler interface {
+	plugin.IPluggable
+
+	// GetTrustAnchorIDs returns an array of trust anchor identifiers used
+	// to retrieve the trust anchors associated with this token. The trust anchors may be necessary to validate the
+	// entire token and/or extract its claims (if it is encrypted).
+	GetTrustAnchorIDs(token *proto.AttestationToken) ([]string, error)
+
+	// SynthKeysFromRefValue synthesizes lookup key(s) for the
+	// provided reference value endorsement.
+	SynthKeysFromRefValue(tenantID string, refVal *Endorsement) ([]string, error)
+
+	// SynthKeysFromTrustAnchor synthesizes lookup key(s) for the provided
+	// trust anchor.
+	SynthKeysFromTrustAnchor(tenantID string, ta *Endorsement) ([]string, error)
+}
```

Comment:
shouldn't we also have a twin `GetRefValueID`?

Reply:
`GetTrustAnchorID` is required, as even before vts can extract the `claims` it can verify Token Integrity using TA which is fetched using `GetTrustAnchorID`.

We do not have a similar API need in the Reference Value side, as Reference Value Keys as on Provisioning Side `keys` are synthesised using RefValues & on Verification Path, we have all the Claims to generate the ReferenceID within the `scheme`  itself and set in `EvidenceContext` to fetch the required RefValues to be used in Appraisal!

Reply:
Sure, but isn't that a bit untidy now that we have clarified the keys/IDs interface?

Reply:
I am happy to consider if there is a suitable use case and a need for it! I just do not want to create `symmetry` for the sake of doing it! I will explore this in a bit more detail!

Reply:
Looking at it, TAID is needed in `vts` generic pipeline code : while the same is not true for RefValIDs. They are strictly not needed in the `vts` code we just set them receiving from one handler and passing them to another `handler` via a variable!  And thinking a bit more, one change is needed, is that we should rename API `ExtractClaims()` to 
`ExtractClaimsAndReferenceIDs()`

Reply:
See issue #213

Reply:
> See issue #213

And given the fact that ReferenceID's are constructed using Claims, there is no need to split the Claims Extraction and ReferenceID generation, a split here does not make sense!

Reply:
As per our conversation raised https://github.com/veraison/services/issues/214 to discuss (first) and then `implement` potential alignment